### PR TITLE
Backport PRs #8374 and its follow-ups #8888 and #8758 into 2.2

### PR DESCRIPTION
**Pull request by prlanzarin**
### What does this PR do?

This is a backport of PR #8374 and its follow-ups #8888, #8758 from the develop branch 
into the 2.2-release branch.

This essentially backports the video-provider from 2.3 into 2.2. Feature-wise, this
relates to the ability to share multiple cameras concurrently per session, but that's
disabled by default. Code-wise, this is a backport of a mild refactor and stability improvements 
done to the video-provider.

This code has already experienced **production environments** for months now so it is good. **Additional testing is needed,
though, to make sure I didn't mess anything up when backporting.** Testing hands are appreciated.

Side-effects:
  - Removal of the old, 2.0 based video stats handling and UI (#5422). That never worked properly,
was never actually used in production and was just bloating the code tree, so it was removed
and I don't see how this would be a breaking change for 2.2.
  - Partial removal of the other old, 2.2 based video stats and network information (#7101), for similar reasons: crippled with false positives, lack of widespread usage. 2.3 has a battle hardened substitute for both.
  - Removal of the old, 2.0 based visibility monitor used for pausing video streams when there was no focus into the conference window. This has long been deactivated in the server due to some operational usage issues and the code was still lingering in the client.
  - PRs #8888 and #8758 commits **will be duplicated** in the 2.3 tree. I could not avoid that because there was no way to cleanly backport them. There was no feature branch that was compatible with the 2.2 tree, so I had to cherry-pick. PR #8374 was cleanly backported so it won't be duplicated.

### Closes Issue(s)
None

### Motivation

Apart from the stability improvements it brings, this is a groundwork PR which will
serve as basis for **subsequent** PRs made to 2.2 in regards to the video-provider.
In the short term, those subsequent changes are:
  - Dynamic, client-side bitrate and frame rate throttling based on number of streams
and
  - UI updates
  - Camera publishing limiting
  - Camera subscription limiting

### CODE REVIEW

File: bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js [hunk 1]
```diff
@@ -46,7 +45,6 @@ const currentParameters = [
   'bbb_preferred_camera_profile',
   'bbb_enable_screen_sharing',
   'bbb_enable_video',
-  'bbb_enable_video_stats',
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js:49]: Note to self -  remove from the documentation

File: bigbluebutton-html5/private/locales/en.json [hunk 2]
```diff
@@ -546,6 +546,9 @@
     "app.videoPreview.closeLabel": "Close",
     "app.videoPreview.findingWebcamsLabel": "Finding webcams",
     "app.videoPreview.startSharingLabel": "Start sharing",
+    "app.videoPreview.stopSharingLabel": "Stop sharing",
+    "app.videoPreview.stopSharingAllLabel": "Stop all",
+    "app.videoPreview.sharedCameraLabel": "This camera is already being shared",
```
Comment (antobinary) [inline bigbluebutton-html5/private/locales/en.json:551]: I think a bunch of the ` "app.video.*` messages can be removed now

File: bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js (hunk 1 above)
Comment (antobinary) [reply bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js:49]: Did it! https://github.com/bigbluebutton/bigbluebutton.github.io/commit/0a5fbf9b395946aaddc9deafb84d29f1d3f21b05

**Comment by uka-support** [comment]
I noticed that my own webcam is not the first any more - all images are sorted alphabetically now. Is that intenional?

**Comment by prlanzarin** [comment]
Yes.

**Comment by uka-support** [comment]
That's rather counter-intuitive. Most conferencing solutions put your own image first, so you can easily spot yourself and check your makeup ;-)

What's the reason behind that decision?

**Comment by prlanzarin** [comment]
Technical. The way that reordering was done before was quite expensive and convoluted, and some code rearrangement removed it.

I get your point (and I agree, partially). The reordering might still come back in 2.2 if I find a clean way to do it.

**Comment by uka-support** [comment]
I'd appreciate if the original order came back.

Thanks for the great work! 👍

**Comment by prlanzarin** [comment]
@uka-support original stream sorting was restored in #10318 (to be included in 2.2.23).

**Pull request merged**